### PR TITLE
Make MicrometerCollector implement Collector.Describable

### PR DESCRIPTION
**Pull request by izeye**
This PR changes to make `MicrometerCollector` implement `Collector.Describable`.

Closes gh-2087

### CODE REVIEW

**Comment by izeye** [comment]
For the record, I chose to use `Type.HISTOGRAM` unconditionally as it includes all the names of `Type.SUMMARY`.

File: implementations/micrometer-registry-prometheus/src/main/java/io/micrometer/prometheus/MicrometerCollector.java [hunk 1]
```diff
@@ -77,6 +78,37 @@ public boolean isEmpty() {
...
+                return Collections.singletonList(
+                        new MetricFamilySamples(conventionName, Type.COUNTER, help, Collections.emptyList()));
+
+            case GAUGE:
+                return Collections.singletonList(
+                        new MetricFamilySamples(conventionName, Type.GAUGE, help, Collections.emptyList()));
+
+            case TIMER:
+                return Collections.singletonList(
+                        new MetricFamilySamples(conventionName, Type.HISTOGRAM, help, Collections.emptyList()));
```
Comment (shakuzen) [inline implementations/micrometer-registry-prometheus/src/main/java/io/micrometer/prometheus/MicrometerCollector.java:94]: Also need the max GAUGE like the distribution summary

Comment (shakuzen) [reply implementations/micrometer-registry-prometheus/src/main/java/io/micrometer/prometheus/MicrometerCollector.java:94]: We should increase test cases for filtering by time series name to ensure we're not missing any like this. For example, with a Timer named `my.timer` and attempting to filter by `my_timer_duration_seconds_max` (`PrometheusMeterRegistryTest` uses the `PrometheusDurationNamingConvention` by default) does not work as is right now.

Comment (izeye) [reply implementations/micrometer-registry-prometheus/src/main/java/io/micrometer/prometheus/MicrometerCollector.java]: @shakuzen Thanks for pointing it out! I changed as suggested.

**Pull request merged**